rpcserver.go

```diff
@@ -1372,6 +1372,23 @@ func (r *rpcServer) savePayment(route *routing.Route, amount lnwire.MilliSatoshi
 	return r.server.chanDB.AddPayment(payment)
 }
 
+// validatePayReqExpiry checks if the passed payment request has expired. In
+// the case it has expired, an error will be returned.
+func validatePayReqExpiry(payReq *zpay32.Invoice) error {
+	// If no expiry is set in the payment request, it implies a 3600
+	// second expiry time.
+	expiry := 3600 * time.Second
+	if payReq.Expiry != nil {
+		expiry = time.Duration(payReq.Expiry.Unix()) * time.Second
```

Comment:
`Unix` returns of the number of _seconds_ since January 1st 1970. `time.Duration` itself defaults to units of `ns` if it isn't scaled at all. As a result, I believe this computation should actually be: 
```go
time.Second(payReq.Expiry.Unix())
```

Otherwise, this will always scale up the expiration time in seconds by a few orders of magnitude.

Reply:
This aside, I like the suggestion of instead natively moving the expiration validation within `zpay32` itself (along with all the other invoice related validation).